internal/database/client.go

```diff
@@ -0,0 +1,79 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+	hnModel "github.com/emmaLP/gs-software-onboarding/pkg/hackernews/model"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+)
+
+type Client interface {
+	SaveItem(item *hnModel.Item) error
+	CloseConnection()
+}
+
+type database struct {
+	mongoClient  *mongo.Client
+	context      context.Context
+	logger       *zap.Logger
+	databaseName string
+}
+
+const mongoUriTemplate = "mongodb://%s:%s"
+
+func New(ctx context.Context, logger *zap.Logger, config *model.DatabaseConfig) (*database, error) {
+	mongoUri := fmt.Sprintf(mongoUriTemplate, config.Host, config.Port)
+
+	opts := options.Client().ApplyURI(mongoUri)
+	if strings.TrimSpace(config.Username) != "" && strings.TrimSpace(config.Password) != "" {
+		credentials := options.Credential{
+			Username: config.Username,
+			Password: config.Password,
+		}
+		opts = opts.SetAuth(credentials)
+	}
+	client, err := mongo.Connect(ctx, opts)
```

Comment:
Might be worth checking to see if the database is ready. you can do something like this... 

```
	for {
		select {
		case <-ctx.Done():
			return nil, fmt.Errorf("mongo refused to connect: %v %w", ctx.Err(), err)
		default:
			err := client.Ping(ctx, readpref.Primary())
			if err == nil {
				log.Print("mongo is now connected")
				return client, nil
			}
		}
	}
```